tests/check/check_h26xsigned_auth.c

```diff
@@ -1787,76 +1786,76 @@ static void validation_helper_function(signed_video_t *sv, nalu_list_item_t *sei
+    // starting the session.
+    {true, true, false, true},
   };
 
-  for (int j = 0; j < (int) (sizeof(pk_tests) / sizeof(*pk_tests)); j++) {
+  for (size_t j = 0; j < sizeof(pk_tests) / sizeof(*pk_tests); j++) {
     SignedVideoReturnCode sv_rc;
     SignedVideoCodec codec = settings[_i].codec;
-    nalu_list_item_t *i_nalu = nalu_list_item_create_and_set_id("I", 0, codec);
     nalu_list_item_t *sei = NULL;
     signed_video_t *sv_camera = NULL;
     sign_algo_t algo = settings[_i].algo;
     char *private_key = NULL;
 
     sv_camera = generate_and_set_private_key_on_camera_side(settings[_i], pk_tests[j].pk_in_sei,
-        i_nalu, &sei);
+        &sei);
 
     // On validation side
     signed_video_t *sv_vms = signed_video_create(codec);
 
     signature_info_t sign_info_wrong_key = {0};
-    if (pk_tests[j].wrong_pk) {
-      // Generate a new private key in order to extract a bad private key (a key not compatible with
-      // the one generated on the camera side)
-      signed_video_generate_private_key( algo, "./", (char **)&sign_info_wrong_key.private_key,
-          &sign_info_wrong_key.private_key_size);
-      openssl_read_pubkey_from_private_key(&sign_info_wrong_key);
-      // Set public key
-      sv_rc = signed_video_set_public_key(sv_vms, sign_info_wrong_key.public_key,
-          sign_info_wrong_key.public_key_size); ck_assert_int_eq(sv_rc, SV_OK);
+    // Generate a new private key in order to extract a bad private key (a key not compatible with
+    // the one generated on the camera side)
+    signed_video_generate_private_key( algo, "./", (char **)&sign_info_wrong_key.private_key,
+        &sign_info_wrong_key.private_key_size);
+    openssl_read_pubkey_from_private_key(&sign_info_wrong_key);
+
+    signature_info_t *sign_info = sv_camera->signature_info;
+    if (pk_tests[j].use_wrong_pk) {
+      sign_info = &sign_info_wrong_key;
     }
-
-    if (pk_tests[j].set_pk) {
+    if (pk_tests[j].set_pk_before_session_start) {
       // Set public key
-      sv_rc = signed_video_set_public_key( sv_vms, sv_camera->signature_info->public_key,
-          sv_camera->signature_info->public_key_size); ck_assert_int_eq(sv_rc, SV_OK);
+      sv_rc = signed_video_set_public_key( sv_vms, sign_info->public_key,
+          sign_info->public_key_size);
+      ck_assert_int_eq(sv_rc, SV_OK);
     }
-
-    signature_info_t *sign_info = NULL;
-    if (pk_tests[j].use_signature_info) {
-      if (pk_tests[j].wrong_pk) {
-        sign_info = &sign_info_wrong_key;
-      } else {
-        sign_info = sv_camera->signature_info;
-      }
+    if (!pk_tests[j].set_pk_after_session_start) {
+      sign_info = NULL;
     }
-    validation_helper_function(sv_vms, sei, pk_tests[j].wrong_pk, sign_info, codec);
+    validate_public_key_scenario(sv_vms, sei, pk_tests[j].use_wrong_pk, sign_info, codec);
 
     signed_video_free(sv_camera);
     free(private_key);
```

Comment:
You should free `sv_vms`and `sei` here now when it is not done inside `validate_public_key_scenario`